### PR TITLE
[script.sublissimo] 0.0.2

### PR DESCRIPTION
**Pull request by Allertj**
- New: frame rate conversion on the fly
 - New: play modified subtitle directly on video
 - Minor bugfixes

### Checklist:

- [X] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [X] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-scripts/blob/master/CONTRIBUTING.md) document
- [X] Each add-on submission should be a single commit with using the following style: [script.foo.bar] 1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practise but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-scripts/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.

### CODE REVIEW

File: script.sublissimo/resources/lib/play_along_file.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+             xbmcvfs.delete(temp_file)
+        if pause:
+            self.proper_exit = True
+            xbmc.Player().stop()
+
+    def activate_sub(self):
+        new_file_name = self.filename[:-4] + "_temp.srt"
+        with closing(File(new_file_name, 'w')) as fo:
+            fo.write("".join(self.subtitlefile))
+        xbmc.Player().setSubtitles(new_file_name)
```
Comment (enen92) [inline script.sublissimo/resources/lib/play_along_file.py]: self

File: script.sublissimo/resources/lib/play_along_file.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+                self.delete_temp_file(True)
+                script.save_the_file(self.subtitlefile, self.filename)
+            if choice == 4:
+                self.delete_temp_file(True)
+                script.show_dialog(self.subtitlefile, self.filename)
+            if choice == 5:
+                self.delete_temp_file(True)
+                script.exiting(self.subtitlefile, self.filename)
+            if not self.flag:
+                xbmc.Player().pause()
```
Comment (enen92) [inline script.sublissimo/resources/lib/play_along_file.py]: self

File: script.sublissimo/resources/lib/sync_by_frame_rate.py [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+    def write_and_display_temp_file(self, new_subtitlefile, temp):
+        if temp:
+            new_file_name = self.filename[:-4] + "_temp.srt"
+        else:
+            self.delete_temp_file()
+            new_file_name = self.filename[:-4] + "_edited.srt"
+        with closing(File(new_file_name, 'w')) as fo:
+            fo.write("".join(new_subtitlefile))
+        self.new_subtitlefile = new_subtitlefile
+        xbmc.Player().setSubtitles(new_file_name)
```
Comment (enen92) [inline script.sublissimo/resources/lib/sync_by_frame_rate.py]: self

File: script.sublissimo/resources/lib/sync_by_frame_rate.py [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+            self.rearrange(chosen_factor, from_pause)
+        if menuchoice == 6:
+            self.get_frame_rate()
+        if menuchoice == 7:
+            xbmcgui.Dialog().ok(_(32114), _(32115))
+            response = xbmcgui.Dialog().input(_(32113))
+            calculated_factor = eval(str(response))
+            self.rearrange(calculated_factor, from_pause)
+        if menuchoice == 8 or menuchoice == -1:
+            xbmc.Player().stop()
```
Comment (enen92) [inline script.sublissimo/resources/lib/sync_by_frame_rate.py]: self

**Comment by Allertj** [comment]
Thanks, its fixed...

**Comment by Allertj** [comment]
Could you possibly wait a bit before merging? There is a small thing that needs to be fixed...

**Pull request merged**